### PR TITLE
Validation phase (ResNets training)

### PR DESCRIPTION
**Pull request by Zaltad**
I think this "validation" part aligns more with paper's implementation see train1 and validate1 from https://github.com/suakaw/BDQ_PrivacyAR/blob/main/action-recognition-pytorch-entropy/utils/utils.py#L318 . And this https://github.com/suakaw/BDQ_PrivacyAR/blob/main/action-recognition-pytorch-entropy/train.py#L427 (TARGET MODEL TRAINING and BUDGET MODEL TRAINING). Although authors claim in the paper that they average softmax outputs across frames, apparently they adopt different method in the implementation (take random frame), only sign of possible mean usage is https://github.com/suakaw/BDQ_PrivacyAR/blob/main/action-recognition-pytorch-entropy/models/twod_models/resnet.py#L220 , but I wasn't able to identify if this ResNet is used anywhere, all models pointed to different ResNets implementations https://github.com/suakaw/BDQ_PrivacyAR/blob/main/action-recognition-pytorch-entropy/models/threed_models/i3d.py#L19 .
It would be nice to somehow refactor the train_once_... and ..._training functions, but I don't know if that would be more readable.

### CODE REVIEW

**Comment by cactusutcac** [comment]
We did a train/val/test split in preprocessing. However, the paper never mentioned any test set. The validation set is used as the evaluation set. Do we need to change our dataset-splitting strategy to better align with the paper? But as discussed yesterday, the hardware limit makes it difficult to use 2 clips for training so we are using only 1 for training and 1 for validation. Maybe we better stick to our current 1-train / 1-val setup and ignore the test set for now
